README.md

```diff
@@ -91,21 +95,102 @@ api_key: your-evergreen-api-key
 3. Generate or copy your API key
 4. Use your username and the generated API key in the configuration file
 
+### Project Configuration (Optional)
+
+**Good news! Project configuration is optional.** The AI agent can determine the project from context naturally.
+
+#### Agent-Driven Approach (Recommended)
+
+AI agents automatically understand your project context through:
+
+1. **Automatic Memory** - When a project is detected, the agent receives it in context
+2. **Natural Language** - Simply mention your project: "Check my MMS patches"
+3. **Response Context** - All tool responses include project information
+4. **Conversation Memory** - Tell the agent once: "My project is mongodb" and it remembers
+
```

Comment:
[Curious quetion] Do we need that in readme? As this chapter describes MCP arguments processing the same happens with other arguments for the tools we have.